src/libraries/System.Private.CoreLib/src/System/Text/Latin1Utility.cs

```diff
@@ -0,0 +1,1097 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.X86;
+using Internal.Runtime.CompilerServices;
+
+#pragma warning disable SA1121 // explicitly using type aliases instead of built-in types
+#if TARGET_64BIT
+using nint = System.Int64;
+using nuint = System.UInt64;
+#else // TARGET_64BIT
+using nint = System.Int32;
+using nuint = System.UInt32;
+#endif // TARGET_64BIT
+
+namespace System.Text
+{
+    internal static partial class Latin1Utility
+    {
+#if DEBUG
+        static Latin1Utility()
+        {
+            Debug.Assert(sizeof(nint) == IntPtr.Size && nint.MinValue < 0, "nint is defined incorrectly.");
```

Comment:
We define nint in dozens of places without asserts like this. What makes this one special?

Reply:
This same assert was present in `AsciiUtility`. IIRC the reason I added it in the first place is that I saw another project (compiled as AnyCPU) try to reference the source file, which would have led to catastrophe at runtime. I'm perhaps just being overly paranoid. :(

Reply:
Thankfully we can get rid of these once `nint` is in roslyn master 😄